File: code/string_algorithms/src/reverse_word_string/reverse_word_string.js
/*

// string algorithms | reverse word string | reverse word string | JAVASCRIPT
// Part of Cosmos by OpenGenus Foundation

*/


function reverseString(string) {
    const reversedString = string.split(' ').reverse().join(' ');
    console.log(reversedString);
}

reverseString('I know what you did last summer'); // summer last did you what know I
reverseString('OpenGenus cosmos'); // cosmos OpenGenus

File: code/string_algorithms/src/reverse_word_string/reverse_word_string.go
/// Part of Cosmos by OpenGenus Foundation
/// Receives a string and returns the reverse of it
/// Contributed by: Joao Pedro Campos Silva (joaopedrocampos)

package main

import (
	"fmt"
)

// Function to reverse a string
func reverseString(s string) string {
	runes := []rune(s)

	for i, j := 0, len(runes)-1; i < j; i, j = i+1, j-1 {
		runes[i], runes[j] = runes[j], runes[i]
	}

	return string(runes)
}

func main() {
	fmt.Println(reverseString("I know what you did last summer")) // Output: remmus tsal did uoy tahw wonk I
	fmt.Println(reverseString("OpenGenus cosmos"))                // Output: somsoc suneGnepO
	fmt.Println(reverseString("GoLang"))                          // Output: gnaLoG
	fmt.Println(reverseString("12345"))                           // Output: 54321
}
